internal/cmd/egctl/config_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package egctl
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	kube "github.com/envoyproxy/gateway/internal/kubernetes"
+	netutil "github.com/envoyproxy/gateway/internal/utils/net"
+)
+
+var _ kube.PortForwarder = &fakePortForwarder{}
+
+type fakePortForwarder struct {
+	responseBody []byte
+	localPort    int
+	l            net.Listener
+	mux          *http.ServeMux
+}
+
+func newFakePortForwarder(b []byte) (kube.PortForwarder, error) {
+	p, err := netutil.LocalAvailablePort()
+	if err != nil {
+		return nil, err
+	}
+
+	fw := &fakePortForwarder{
+		responseBody: b,
+		localPort:    p,
+		mux:          http.NewServeMux(),
+	}
+	fw.mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		_, _ = w.Write(fw.responseBody)
+	})
+
+	return fw, nil
+}
+
+func (fw *fakePortForwarder) Start() error {
+	l, err := net.Listen("tcp", fw.Address())
+	if err != nil {
+		return err
+	}
+	fw.l = l
+
+	go func() {
+		// nolint: gosec
+		if err := http.Serve(l, fw.mux); err != nil {
+			log.Fatal(err)
+		}
+	}()
+
+	return nil
+}
+
+func (fw *fakePortForwarder) Stop() {}
+
+func (fw *fakePortForwarder) Address() string {
+	return fmt.Sprintf("localhost:%d", fw.localPort)
+}
+
+func TestExtractConfigDump(t *testing.T) {
+	fw, err := newFakePortForwarder(readConfig("in.json"))
+	assert.NoError(t, err)
+	err = fw.Start()
+	assert.NoError(t, err)
+
+	cases := []struct {
+		output   string
+		expected string
+	}{
+		{
+			output:   "json",
+			expected: "out.json",
+		},
+		{
+			output:   "yaml",
+			expected: "out.yaml",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.output, func(t *testing.T) {
+			got, err := extractConfigDump(fw, tc.output)
```

Comment:
Thanks for your effort to put UT around this feature. But in my view, this kind of testes could be better to add as e2e tests. 

In this approach, we put configDumps into fake server, and request to it and get configDumps back, and compare them. It make less sence I think 😄. And we will contain long test jsons/yamls to mimic real envoyproxy configs. cc @arkodg WDYT.

Reply:
it's useful for `config envoy-proxy cluster`

Reply:
@Xunzhuo yeah e2e would be ideal, maybe we can raise a GH issue to track it, but for now, this approach add coverage for a sufficient amount of code in this PR, which Im fine with

Reply:
Thanks @zirain, @arkodg. It makes sense.